tests/dtls-client.c

```diff
@@ -473,40 +478,53 @@ main(int argc, char **argv) {
       }
       break;
     case 'p' :
-      strncpy(port_str, optarg, NI_MAXSERV-1);
-      port_str[NI_MAXSERV - 1] = '\0';
+      local_port = atoi(optarg);
       break;
     case 'r' :
       force_renegotiation_info = 1;
       break;
     case 'v' :
       log_level = strtol(optarg, NULL, 10);
       break;
+    case -1 :
+      /* handle arguments */
+      if (!dst.size) {
+        /* resolve destination address where server should be sent */
+        res = resolve_address(argv[optind++], &dst.addr.sa);
```

Comment:
I'm not convinced that this is the best way do do this (very unusual), moving the logic processing the arguments to within the `getopt()` logic, and fail to see why it is actually needed.  If this is done, at a minimum `dtls_set_log_level(log_level);` needs to be included at the start.

I think that simply removing the use of `port_str`, using the new `local_port` or `DEFAULT_PORT `in the appropriate places along with the other changes should should work

Reply:
I guess, the comment is for PR #221.

This PR and PR #221 are changing the very same lines so I decided to put the changes of this PR on the top of PR #221.

Reply:
Ah - yes, this really should be against #221.

Reply:
I already copied your comment ;-)